src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -140,26 +147,49 @@ protected override void ProcessRecord()
 
                 if (_jschema != null)
                 {
-                    var errorMessages = _jschema.Validate(parsedJson);
-                    if (errorMessages != null && errorMessages.Count != 0)
-                    {
-                        result = false;
-
-                        Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema);
-
-                        foreach (var message in errorMessages)
+                        var validationResults = _jschema.Validate(parsedJson, new ValidationOptions { OutputFormat = OutputFormat.Basic });
+                        result = validationResults.IsValid;
+                        if (!result)
                         {
-                            ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema", ErrorCategory.InvalidData, null);
-                            errorRecord.ErrorDetails = new ErrorDetails(message.ToString());
-                            WriteError(errorRecord);
+                            Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema);
+
+                            if (validationResults.Message != null)
+                            {
+                                ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema", ErrorCategory.InvalidData, null);
+                                var message = $"{validationResults.Message} at {validationResults.InstanceLocation}";
+                                errorRecord.ErrorDetails = new ErrorDetails(message);
+                                WriteError(errorRecord);
+                            }
+
+                            if (validationResults.HasNestedResults)
+                            {
+                                foreach (var nestedResult in validationResults.NestedResults)
+                                {
+                                    if (nestedResult.Message == null)
+                                    {
+                                        continue;
+                                    }
+
+                                    ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema", ErrorCategory.InvalidData, null);
+                                    var message = $"{nestedResult.Message} at {nestedResult.InstanceLocation}";
+                                    errorRecord.ErrorDetails = new ErrorDetails(message);
+                                    WriteError(errorRecord);
+                                }
+                            }
                         }
-                    }
                 }
             }
+            catch (JsonSchemaReferenceResolutionException jsonExc)
+            {
+                result = false;
+
+                Exception exception = new(TestJsonCmdletStrings.InvalidJsonSchema, jsonExc);
+                WriteError(new ErrorRecord(exception, "InvalidJsonSchema", ErrorCategory.InvalidData, _jschema));
+            }
```

Comment:
Have we a test for the branch?

Reply:
All of the cases where a referenced schema fails to load (and there are several) will flow through this `catch`.

Reply:
I don't see tests where we looks JsonSchemaReferenceResolutionException

Reply:
We don't need to. That's an internal implementation detail. It's presented to the user as "invalid JSON schema."

Reply:
We throw InvalidJsonSchema in 3 places - for file, for text and if load from a reference - we haven't a test for the last. So we could create a string with valid schema but with bad reference (to non-existent file) and in InvalidJsonSchema exception check inner exception name for JsonSchemaReferenceResolutionException.

Reply:
That would just be adding an assertion to an existing test. I can do that, but I don't think that it warrants new tests.

Reply:
Do you have an example where inner exceptions are being checked in powershell?  I only see assertions on the error output.

```ps
$errorVar[0].FullyQualifiedErrorId | Should -BeExactly "InvalidJsonAgainstSchema,Microsoft.PowerShell.Commands.TestJsonCommand"
```

This doesn't mention the exceptions that were thrown.

Reply:
> We throw InvalidJsonSchema in 3 places - for file, for text and if load from a reference - we haven't a test for the last.

`Test-Json throw if a schema from file is invalid` which throws an error because _invalid_schema_definitions.json_ is invalid JSON is failing to load a reference.  Granted, it's failing to resolve the reference because the referenced file is invalid JSON, but the reference resolution still fails, triggering a `JsonSchemaReferenceResolutionException`, which is caught on line 186 and repackaged as an "InvalidJsonSchema" error for the user.